### PR TITLE
Implement Binding to Unix Domain Sockets

### PR DESCRIPTION
**Pull request by lorenzleutgeb**
This should be slightly faster than TCP over the loopback interface.

I'd appreciate if someone could help me with the test. The systemd service is locked down, and I am not sure which options I can/should disable to allow access to the socket.

### CODE REVIEW

File: tests/t03-varnish-uds.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+              }
+            '';
+          };
+
+          networking.firewall.allowedTCPPorts = [ 80 ];
+          environment.systemPackages = [ pkgs.hello ];
+
+          systemd.sockets.harmonia-dev = {
+            listenStreams = [ sock ];
+            requiredBy = "harmonia-dev.service";
```
Comment (Mic92) [inline tests/t03-varnish-uds.nix:31]:
```suggestion
            requiredBy = ["harmonia-dev.service"];
```

**Comment by Mic92** [comment]
@mergify rebase

**Comment by Mic92** [comment]
Please re-open if you plan to work further on it.

**Comment by lorenzleutgeb** [comment]
I cannot re-open a closed PR, AFAIK only maintainers can do this. And yes, I would still like to work on this. I'd appreciate help with the systemd service to allow using Harmonia via socket.

**Comment by Mic92** [comment]
I think you can also open a new pull request.

**Pull request closed without merging**